frontend/javascripts/oxalis/model/sagas/load_histogram_data_saga.ts

```diff
@@ -1,88 +1,99 @@
 
-async function fetchAllHistogramsForLayers(
-  dataLayers: Array<DataLayer>,
-  dataset: APIDataset,
-): Promise<HistogramDataForAllLayers> {
-  const histograms: HistogramDataForAllLayers = {};
+export default function* loadHistogramDataSaga(): Saga<void> {
+  yield* take("WK_READY");
+  yield* takeEvery("RELOAD_HISTOGRAM", reloadHistogramForLayer);
+  const dataLayers: Array<DataLayer> = yield* call([Model, Model.getColorLayers]) as any;
+  const layerConfigurations = yield* select((state) => state.datasetConfiguration.layers);
 
+  const histograms: HistogramDataForAllLayers = {};
   for (const dataLayer of dataLayers) {
-    try {
-      // We send the histogram data requests sequentially so there is less blockage of bucket data requests.
-      // eslint-disable-next-line no-await-in-loop
-      const data = await getHistogramForLayer(dataset.dataStore.url, dataset, dataLayer.name);
-
-      if (Array.isArray(data) && data.length > 0) {
-        histograms[dataLayer.name] = data;
-      }
-    } catch (e) {
-      console.warn(`Error: Could not fetch the histogram data for layer ${dataLayer.name}.`, e);
-    }
+    const layerName = dataLayer.name;
+    // We send the histogram data requests sequentially so there is less blockage of bucket data requests.
+    const histogram = yield* call(loadHistogramForLayer, layerName, layerConfigurations[layerName]);
+    if (histogram != null) histograms[layerName] = histogram;
   }
 
-  return histograms;
+  yield* put(setHistogramDataAction(histograms));
 }
 
-export default function* loadHistogramDataSaga(): Saga<void> {
-  yield* take("WK_READY");
-  // Flow does not understand that Array<DataLayer> is returned for some reason.
-  const dataLayers: Array<DataLayer> = yield* call([Model, Model.getColorLayers]) as any;
-  const dataset = yield* select((state) => state.dataset);
+function* reloadHistogramForLayer(action: ReloadHistogramAction): Saga<void> {
+  const { layerName } = action;
   const layerConfigurations = yield* select((state) => state.datasetConfiguration.layers);
-  const histograms = yield* call(fetchAllHistogramsForLayers, dataLayers, dataset);
 
-  for (const layerName of Object.keys(histograms)) {
-    // Adjust the intensityRange of the layer to be within the range of the actual (sampled) data.
-    const histogram = histograms[layerName];
-    const allMinValues = histogram.map((currentHistogramData) => currentHistogramData.min);
-    const allMaxValues = histogram.map((currentHistogramData) => currentHistogramData.max);
-    const minimumInHistogramData = Math.min(...allMinValues);
-    const maximumInHistogramData = Math.max(...allMaxValues);
-    let newIntensityRange = [];
-    const currentLayerConfig = layerConfigurations[layerName];
+  const histogram = yield* call(loadHistogramForLayer, layerName, layerConfigurations[layerName]);
+  // Also update histogram data if it is undefined since apparently it can no longer be generated
```

Comment:
I don't understand the "since apparently it can no longer be generated" bit :thinking:

Reply:
Yes that didn't make too much sense, I removed it.